### PR TITLE
Do not set apm agent instrument setting

### PR DESCRIPTION
**Pull request by pgomulka**
we should not be setting apm agent instrument setting as ES is not using auto instrumentation
this commit removes the possibility to set this and relies on a static settings in ApmJavaOptions

this commit also migrates deprecated setting in RunTask

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against main? Unless there is a good reason otherwise, we prefer pull requests against main and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

File: build-tools/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java [hunk 1]
```diff
@@ -214,7 +214,7 @@ public void beforeStart() {
                 else if (node.getSettingKeys().contains("telemetry.metrics.enabled") == false) { // metrics
                     node.setting("telemetry.metrics.enabled", "false");
                 } else if (node.getSettingKeys().contains("telemetry.tracing.enabled") == false
-                    && node.getSettingKeys().contains("tracing.apm.enabled") == false) { // tracing
+                    && node.getSettingKeys().contains("telemetry.tracing.enable") == false) { // tracing
```
Comment (mosche) [inline build-tools/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java:217]: Good point, though we have to fix the logic differently to support the fallbacks. You are checking "telemetry.tracing.enable" twice this way

Comment (pgomulka) [reply build-tools/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java:217]: fair point. this was stupid find replace.
I don't think we have to worry about deprecated tracing.apm in RunTask
This is a 'dev only' tool.

**Comment by mosche** [comment]
Fine if closing this in favor of https://github.com/elastic/elasticsearch/pull/105055?

**Comment by pgomulka** [comment]
> Fine if closing this in favor of https://github.com/elastic/elasticsearch/pull/105055?

sure, I will close

**Pull request closed without merging**